### DISCUSSION TITLE
Searchable repositories

### DISCUSSION THREAD

**Discussion by JoshPaulie** [community/community#8268] [Mobile]
Right now, you have to manually navigate to each file to find specific references or files. A search box would be a massive QoL improvement.

**Comment by eliperkins** [comment]
Thanks for all the feedback @JoshPaulie! I hear you; navigating code within repositories to find just the code you're looking for is tough right now.

I've added your notes about searching for specific references and files to our internal tracking issue about this. Stay tuned!

**Reply by ghost** [reply]
Agreed. I consider this one of the biggest missing features as large repositories of files are barely usable unless the entire directory structure is committed to memory.

**Comment by elisealix22** [comment]
Hi @JoshPaulie, we recently published a new version of GitHub Mobile (v1.133.0 and above) that adds code search to repositories. Would love to hear your thoughts if you try it out! Thank you again for helping us prioritize this feature.

**Reply by JoshPaulie** [reply]
You guys are sweet "thanking" me, but you are the ones with the stellar implementation.

**Reply by ItsIgnacioPortal** [reply]
The search feature appears to work, but queries are made every time you type a character. This causes the app to be very sluggish while you're typing-in your query. Could a short delay be added, so the app waits until the user finishes typing before making the query?

**Reply by bric3** [reply]
How do you search from a repo page.

E.g. Looking this file `platform/lang-impl/src/com/intellij/ui/popup/PopupPositionManager.java`

From Jetbrains/intellij-community repo.

<img width="50%" src="https://github.com/community/community/assets/803621/3e869112-75af-458a-a40e-1fde1e3d1d56">

**Reply by ItsIgnacioPortal** [reply]
> How do you search from a repo page.

You have to browse the files in the repository, and then the magnifying glass icon will appear in the top-menu.

**Reply by bric3** [reply]
Oh I'm looking at a different branch, the search text field is there on the main branch, but doesn't appear on a different branch.
